apps/builder/app/builder/features/ai/operations.tsx

```diff
@@ -0,0 +1,205 @@
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string>();
+
+  const jsx = useStore($jsx);
+
+  return (
+    <form
+      method="POST"
+      action={restAiOperations()}
+      onSubmit={(event) => {
+        event.preventDefault();
+
+        if (isLoading) {
+          return;
+        }
+
+        const formData = new FormData(event.currentTarget);
+
+        const prompt = formData.get("prompt");
+        let data = formData.get("data");
+
+        if (typeof data === "string") {
+          try {
+            data = JSON.parse(data);
+          } catch {
+            /**/
+          }
+        }
+
+        if (typeof prompt !== "string" || typeof data !== "string") {
+          alert("Invalid data");
+          return;
+        }
+
+        const metas = registeredComponentMetasStore.get();
+        const exclude = ["Body", "Slot"];
+
+        const components = [...metas.keys()]
+          .filter((name) => !exclude.includes(name))
+          .map((name) =>
+            name.replace("@webstudio-is/sdk-components-react-radix:", "Radix.")
+          );
+
+        abort.current = new AbortController();
+
+        request<operations.Response>([
+          restAiOperations(),
+          {
+            method: "POST",
+            body: JSON.stringify({
+              prompt,
+              components,
+              jsx: data,
+            }),
+            signal: abort.current.signal,
+          },
+        ])
+          .then((result) => {
+            if (result.success === true) {
+              if (result.type === "json" && result.id === "operations") {
+                applyOperations(result.data);
+              }
+              return;
+            }
+
+            setError(result.data.message);
+          })
+          .finally(() => {
+            abort.current = undefined;
+            setIsLoading(false);
+          });
+
+        setError(undefined);
+        setIsLoading(true);
+      }}
+    >
+      <Label>
+        Prompt
+        <InputField name="prompt" />
+      </Label>
+      <input type="hidden" name="data" value={JSON.stringify(jsx)} />
+      <Button
+        onClick={(event) => {
+          if (isLoading) {
+            event.preventDefault();
+            abort.current?.abort();
+          }
+        }}
+      >
+        {isLoading ? "stop" : "go"}
+      </Button>{" "}
+      {isLoading ? <SpinnerIcon /> : null}
+      {error ? <Text>{error}</Text> : null}
+    </form>
+  );
+};
+
+const $jsx = computed(
+  [
+    selectedInstanceSelectorStore,
+    instancesStore,
+    propsStore,
+    dataSourcesStore,
+    registeredComponentMetasStore,
+    breakpointsStore,
+    stylesStore,
+    styleSourceSelectionsStore,
+  ],
+  (
+    selectedInstanceSelector,
+    instances,
+    props,
+    dataSources,
+    metas,
+    breakpoints,
+    styles,
+    styleSourceSelections
+  ) => {
+    if (selectedInstanceSelector === undefined) {
+      return null;
```

Comment:
Better undefined or empty string

Reply:
we re generally avoiding using null us much as possible, basically using it in cases where there is no other way only, otherwise undefined, in this case you can just 
```suggestion
      return;
```